src/all/kavita/src/eu/kanade/tachiyomi/extension/all/kavita/Kavita.kt

```diff
@@ -1569,10 +1577,22 @@ class Kavita(private val suffix: String = "") : ConfigurableSource, UnmeteredSou
                 val volumeId = chapter.url.substringAfter("/Volume/").substringBefore("?").toIntOrNull()
                     ?: throw IOException("Invalid volume ID")
 
-                // Get all chapters & pages in this volume
+                // Get all pages in this volume
                 val volumeRequest = GET("$apiUrl/Volume/$volumeId", headersBuilder().build())
                 val volume = client.newCall(volumeRequest).execute().parseAs<VolumeDto>()
+                val matchingChapter = volume.chapters.firstOrNull() // or match a chapterId if available
+                    ?: throw IOException(helper.intl["error_no_chapters_found"])
+
                 val pages = mutableListOf<Page>()
```

Comment:
[Nitpick] It might be more clear to rewrite this as:
```
val pages = (0 until matchingChapter.pages).map { i ->
    Page(
        index = i,
        imageUrl = "$apiUrl/Reader/image?chapterId=${matchingChapter.id}&page=$i&extractPdf=true&apiKey=$apiKey"
    )
}
```

Map is a very familiar method (ES6 with Javascript).